common/docker/params.go

```diff
@@ -0,0 +1,42 @@
+package docker
+
+import (
+	"github.com/DataDog/test-infra-definitions/common"
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type Params struct {
+	optionalDockerAgentParams *DockerAgentParams
+	composeEnvVars            map[string]string
+	composeContent            string
+	pulumiResources           []pulumi.ResourceOption
+	commonEnv                 *config.CommonEnvironment
+}
+
+func newParams(commonEnv *config.CommonEnvironment, options ...func(*Params) error) (*Params, error) {
+	return common.ApplyOption(&Params{commonEnv: commonEnv}, options)
+}
+
+func WithComposeContent(content string, env map[string]string) func(*Params) error {
```

Comment:
It may not be the case after handling https://github.com/DataDog/test-infra-definitions/pull/28/files/66d8aa5478aa33a9b1e6aeba9840406c9f6e3ade..9517792f02e8a2718d9baf2e068a016ac07c7ef1#r1071947273.